File: uitest-helpers/ObjC/XCTestCase+PercySnapshot.m
//
//  XCTestCase+PercySnapshot.m
//
//  Created by Miklos Fazekas on 30/10/16.
//  Copyright © 2016 Miklos Fazekas. All rights reserved.
//

#import "XCTestCase+PercySnapshot.h"

@interface XCTestCase()
- (void) startActivityWithTitle:(NSString*) title block:(void (^)(void)) block;
@end

@interface XCUIApplication()
- (void) _waitForQuiescence;
@end

@implementation XCTestCase (PercySnapshot)
- (void) percySnapshotWithPath: (NSString*) path
{
    [self startActivityWithTitle:[NSString stringWithFormat:@"io.percy/%@", path] block:^{
        XCUIApplication* app = [[XCUIApplication alloc] init];
        [app _waitForQuiescence];
    }];
}
@end

File: uitest-helpers/Swift/XCTestCase+percySnapshot.swift
//
//  XCTestCase+percySnapshot.swift
//  ios-ci-sample
//
//  Created by Miklos Fazekas on 30/10/16.
//  Copyright © 2016 Miklos Fazekas. All rights reserved.
//

import Foundation
import XCTest

@objc protocol XCTestCasePrivate {
    typealias ObjCBlock = @convention(block) ()->Void
    @objc func startActivity(withTitle: String!, block:ObjCBlock) -> Void;
}

extension XCTestCase {
    public func percySnapshot(path: String) -> Void {
        (self as AnyObject).startActivity!(withTitle:"io.percy/\(path)", block:{() in
            let app = XCUIApplication()
            app.perform(Selector(("_waitForQuiescence")))
        })
    }
}
